bqskit/runtime/manager.py

```diff
@@ -217,6 +224,7 @@ def send_up_or_schedule_tasks(self, tasks: Sequence[RuntimeTask]) -> None:
         if num_idle != 0:
             self.outgoing.put((self.upstream, RuntimeMessage.UPDATE, num_idle))
             self.schedule_tasks(tasks[:num_idle])
+            self.update_upstream_idle_workers()
```

Comment:
On Submit msgs we do not set num_idle correct? We do not have to update upstream because the upstream now has an accurate number of idle workers bc of read receipts?

Reply:
In this case, the submit was received from below, and we are scheduling it on the manager locally. Therefore, the manager's boss doesn't know about the updated idle count. For example, if we have one job go to the first worker, who spawns 30 other jobs. The manager may decide to schedule that all locally without the server/higher-manager knowing. Therefore, we now have 30 fewer idle workers on this manager and must inform the server. But when we do send the WAITING message upstream, that is tagged with the last submit call we received from the boss, so it can accurately update its idle count for this manager with information that the manager may not have processed yet.